### PR TITLE
fix(eventarc): sets protojson to DiscardUnknown

### PR DESCRIPTION
**Pull request by telpirion**
## Description

This PR sets all of the calls to `protojson.Unmarshal()` to `DiscardUnknown` fields. This will reduce the likelihood of customers encountering errors in their deployed functions. This PR also adds a comment explaining the usage of the `DiscardUnknown` field.

### CODE REVIEW

File: functions/functionsv2/firebase/upper/upper.go [hunk 1]
```diff
@@ -65,7 +65,15 @@ func init() {
...
-	if err := proto.Unmarshal(e.Data(), &data); err != nil {
+
+	// If you omit `DiscardUnknown`, protojson.Unmarshal returns an error
+	// when encountering a new or unknown field.
+	options := proto.UnmarshalOptions{
+		DiscardUnknown: true,
+	}
+	err := options.Unmarshal(e.Data(), &data)
+
+	if err != nil {
```
Comment (NimJay) [inline functions/functionsv2/firebase/upper/upper.go:76]: Thanks for also moving `if err != nil` into its own line.

File: eventarc/storage_handler/handler_test.go [hunk 2]
```diff
@@ -96,31 +96,3 @@ func TestHelloStorage_NotCloudEvent(t *testing.T) {
...
-	if got := w.Result().StatusCode; got != wantStatus {
-		t.Errorf("got %q, want %q", got, wantStatus)
-	}
-
-	// Ensure failure on malformed storage data.
-	wantBody := "Bad Request: expected Cloud Storage event\n"
-	if got := w.Body.String(); got != wantBody {
-		t.Errorf("got %q, want %q", got, wantBody)
-	}
-}
```
Comment (NimJay) [inline eventarc/storage_handler/handler_test.go:126]: **Question:** What was the exact error that prompted the removal of this test case?

**Review by NimJay — APPROVED** [review]
Thanks for doing this! Looks good to me. Left a comment about the removed test.

File: eventarc/storage_handler/handler_test.go (hunk 2 above)
Comment (telpirion) [reply eventarc/storage_handler/handler_test.go:126]: That's a great question. The removed test checked that the sample would throw an error if the handler received a field that it didn't recognize. Unfortunately, this very issue (throwing an error on an unknown field) has been the source of some problems for customers.

File: functions/functionsv2/firebase/upper/upper.go (hunk 1 above)
Comment (telpirion) [reply functions/functionsv2/firebase/upper/upper.go:76]: 👍🏻

**Comment by telpirion** [comment]
> Thanks for doing this! Looks good to me. Left a comment about the removed test.
> 
> Feel free to merge.

Thank you so much, @NimJay !

**Pull request merged**